frontend/src/Components/SearchBar/SearchBar.jsx

```diff
@@ -1,90 +1,82 @@
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect } from "react";
+import debounce from "lodash.debounce";
 import "./SearchBar.css";
 
-const SearchBar = ({ onSelect, onSearchOnMap }) => {
+function SearchBar({ userId, onSelect }) {
   const [query, setQuery] = useState("");
-  const [results, setResults] = useState([]);
-  const [showDropdown, setShowDropdown] = useState(false);
-  const latestRequest = useRef(null);
+  const [suggestions, setSuggestions] = useState([]);
 
-  const handleInputChange = async (e) => {
-    const value = e.target.value;
-    setQuery(value);
+  const fetchSuggestions = debounce(async (input) => {
+    if (input.length > 0) {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/search?query=${input}&userId=${userId}`
+        );
 
-    if (value.trim() === "") {
-      setResults([]);
-      setShowDropdown(false);
-      return;
-    }
-
-    if (latestRequest.current) {
-      latestRequest.current.abort();
-    }
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
```

Comment:
nit: including the HTTP status code in the error could be helpful for future debugging

Reply:
You're right! Done.